src/creature.cpp

```diff
@@ -506,15 +507,15 @@ void Creature::onDeath()
 	}
 
 	for (const auto& it : experienceMap) {
-		it.first->onGainExperience(it.second, this);
+		it.first->onGainExperience(it.second, getCreature());
 	}
 
 	if (mostDamageCreature) {
 		if (mostDamageCreature != lastHitCreature && mostDamageCreature != lastHitCreatureMaster) {
-			Creature* mostDamageCreatureMaster = mostDamageCreature->getMaster();
+			auto mostDamageCreatureMaster = mostDamageCreature->getMaster();
```

Comment:
Could this be `const auto`
```suggestion
			const auto mostDamageCreatureMaster = mostDamageCreature->getMaster();
```